src/java/com/puppetlabs/DirWatchUtils.java

```diff
@@ -49,6 +49,18 @@ public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs)
                     register(watcher, dir);
                     return FileVisitResult.CONTINUE;
                 }
+
+                @Override
+                public FileVisitResult visitFileFailed(Path file, IOException exc)
+                        throws IOException {
+                     if (exc instanceof java.nio.file.NoSuchFileException) {
```

Comment:
There's also a `NotDirectoryException` though I think that would be bizarre as it would mean Java tried to list the directory's children, but the directory was converted to a file or symlink. So 👍 as is.